src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js

```diff
@@ -98,10 +99,16 @@ class MultisigImportanceTransferCtrl {
             this.nodes = this._Nodes.get();
         } else {
             this.nodes = [];
-            // Get supernodes
-            nem.com.requests.supernodes.all().then((data) => {
+            // Get superNodes with pageNumber 0
+            this._superNodeProgram.getNodes(0).then((data) => {
                 this._$timeout(() => {
-                    this.nodes = this._$filter('toEndpoint')(data.nodes);
+                    // shuffle nodes
+                    data.sort(() => Math.random() - 0.5);
+
+                    this.nodes = data.map((node) => {
+                        const { protocol, hostname, port } = new URL(node.endpoint);
+                        return nem.model.objects.create("endpoint")(`${protocol}//${hostname}`, Number(port))
+                    })
```

Comment:
please move this logic to superNodeProgram so it is in single place. you can call it something like getRandomNode(s) and call it from both places